pkg/api/handlers/libpod/containers.go

```diff
@@ -325,3 +326,84 @@ func ListContainerBatch(rt *libpod.Runtime, ctr *libpod.Container, opts shared.P
 	}
 	return ps, nil
 }
+
+func Checkpoint(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Keep           bool   `schema:"keep"`
+		LeaveRunning   bool   `schema:"leaveRunning"`
+		TCPEstablished bool   `schema:"tcpEstablished"`
+		Export         string `schema:"export"`
+		IgnoreRootFS   bool   `schema:"ignoreRootFS"`
+	}{
+		// override any golang type defaults
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+	name := utils.GetName(r)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+	ctr, err := runtime.LookupContainer(name)
+	if err != nil {
+		utils.ContainerNotFound(w, name, err)
+		return
+	}
+	options := libpod.ContainerCheckpointOptions{
+		Keep:           query.Keep,
+		KeepRunning:    query.LeaveRunning,
+		TCPEstablished: query.TCPEstablished,
+		TargetFile:     query.Export,
+		IgnoreRootfs:   query.IgnoreRootFS,
+	}
+	err = ctr.Checkpoint(r.Context(), options)
+	if err != nil {
+		utils.InternalServerError(w, err)
+	}
+	utils.WriteResponse(w, http.StatusOK, entities.CheckpointReport{Id: ctr.ID()})
+}
+
+func Restore(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Keep            bool   `schema:"keep"`
+		TCPEstablished  bool   `schema:"tcpEstablished"`
+		Import          string `schema:"import"`
```

Comment:
Same as Export, but in reverse - the tar is going to be on the client, so we need to stream it to the server, save to a temporary file, and then pass that into libpod Restore.